v2/internal/controllers/crd_aks_fleet_20230315preview_test.go

```diff
@@ -0,0 +1,174 @@
+	flt := &fleet.Fleet{
+		ObjectMeta: tc.MakeObjectMeta("fleet"),
+		Spec: fleet.Fleet_Spec{
+			Location: region,
+			Owner:    testcommon.AsOwner(rg),
+			HubProfile: &fleet.FleetHubProfile{
+				DnsPrefix: to.Ptr("aso"),
+			},
+			Tags: map[string]string{
+				"name": "test-tag",
+			},
+		},
+	}
+	// creating a fleet
+	tc.CreateResourceAndWait(flt)
+	tc.Expect(flt.Status.Id).ToNot(BeNil())
+	tc.Expect(flt.Status.Tags).ToNot(BeNil())
+	tc.Expect(flt.Spec.Tags["name"]).To(Equal("test-tag"))
+	armId := *flt.Status.Id
+
+	// patching a fleet
+	old := flt.DeepCopy()
+	flt.Spec.Tags = map[string]string{
+		"name": "test-tag2",
+	}
+	tc.PatchResourceAndWait(old, flt)
+	tc.Expect(flt.Spec.Tags["name"]).To(Equal("test-tag2"))
+
+	adminUsername := "adminUser"
+	sshPublicKey, err := tc.GenerateSSHKey(2048)
+	tc.Expect(err).ToNot(HaveOccurred())
+
+	osType := aks.OSType_Linux
+	agentPoolMode := aks.AgentPoolMode_System
+	cluster := &aks.ManagedCluster{
+		ObjectMeta: tc.MakeObjectMeta("mc"),
+		Spec: aks.ManagedCluster_Spec{
+			Location:  region,
+			Owner:     testcommon.AsOwner(rg),
+			DnsPrefix: to.Ptr("aso"),
+			AgentPoolProfiles: []aks.ManagedClusterAgentPoolProfile{
+				{
+					Name:   to.Ptr("ap1"),
+					Count:  to.Ptr(1),
+					VmSize: to.Ptr("Standard_DS2_v2"),
+					OsType: &osType,
+					Mode:   &agentPoolMode,
+				},
+			},
+			LinuxProfile: &aks.ContainerServiceLinuxProfile{
+				AdminUsername: &adminUsername,
+				Ssh: &aks.ContainerServiceSshConfiguration{
+					PublicKeys: []aks.ContainerServiceSshPublicKey{
+						{
+							KeyData: sshPublicKey,
+						},
+					},
+				},
+			},
+			Identity: &aks.ManagedClusterIdentity{
+				Type: to.Ptr(aks.ManagedClusterIdentity_Type_SystemAssigned),
+			},
+			OidcIssuerProfile: &aks.ManagedClusterOIDCIssuerProfile{
+				Enabled: to.Ptr(true),
+			},
+		},
+	}
+	tc.CreateResourceAndWait(cluster)
+	tc.Expect(cluster.Status.Id).ToNot(BeNil())
+	clusterArmID := *cluster.Status.Id
+
+	// Run sub tests - updateRun subtest depends on fleetMember subtest
+	tc.RunSubtests(
+		testcommon.Subtest{
+			Name: "Fleet FleetMember CRUD",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				AKS_Fleet_FleetMember_20230315Preview_CRUD(tc, flt, clusterArmID)
+			},
+		},
+		testcommon.Subtest{
+			Name: "Fleet UpdateRun CRUD",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				AKS_Fleet_UpdateRun_20230315Preview_CRUD(tc, flt)
+			},
+		})
+
+	// delete managed cluster
+	tc.DeleteResourceAndWait(cluster)
+
+	// delete a fleet
+	tc.DeleteResourceAndWait(flt)
+
+	// Ensure that fleet was really deleted in Azure
+	exists, retryAfter, err := tc.AzureClient.HeadByID(tc.Ctx, armId, string(fleet.APIVersion_Value))
+	tc.Expect(err).ToNot(HaveOccurred())
+	tc.Expect(retryAfter).To(BeZero())
+	tc.Expect(exists).To(BeFalse())
+}
+
+func AKS_Fleet_FleetMember_20230315Preview_CRUD(tc *testcommon.KubePerTestContext, flt *fleet.Fleet, clusterArmID string) {
+
+	flt_member := &fleet.FleetsMember{
```

Comment:
minor, style: name of variable should be `fltMember` as per standard Go local variable naming guidelines